docs/docs/config.md

```diff
@@ -90,6 +90,13 @@ option                         | argument                            | descripti
 -------------------------------|-------------------------------------| --------------------------------------------------- | -------
 sqs_queue_uri                  | STRING                              | SQS Queue URI |
 
+## sns producer 
+option                         | argument                            | description                                         | default
+-------------------------------|-------------------------------------| --------------------------------------------------- | -------
+sns_topic                      | STRING                              | The SNS topic to publish to. FIFO topics should end with `.fifo` |
+sns_database_attribute         | BOOLEAN                             | Add `database` as an attribute to the SNS message   | false
+sns_table_attribute            | BOOLEAN                             | Add `table` as an attribute to the SNS message      | false
```

Comment:
this interface is a little weird.  can you explain what the function of SNS message attributes are?  In other words, are they just metadata, or do they perform some role in uniqueness, sorting, querying, something?

Reply:
The `database` and `table` (and possibly other) MessageAttributes allow consumers of the SNS topic to filter which messages they receive from SNS. Using this, multiple different handlers could listen to a single maxwell stream through SNS, and only receive events for databases / tables they are interested in

https://docs.aws.amazon.com/sns/latest/dg/sns-subscription-filter-policies.html#example-filter-policies

Reply:
Follow up question to the above. Would it make more sense to maybe use 
```
sns_partition_by_table BOOLEAN 
sns_partition_by_database BOOLEAN
```
to reference these? I don't know if that makes sense, as you could partition by both fields at the same time, but it uses similar syntax to the other producers that support partitioning

Reply:
I was thinking that we might want to support other attributes copied into the record at some point in time. 

Maybe we're not gonna need it, but I was thinking the interface could be like:

```
--sns_attrs=database,table,xid,ts
```

with a default of `database,table`

Reply:
My only hang up with that, after re-reading some of the documentation, is that the message attributes need to be specified as a specific type of value. https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html 

So we would have to split on the string, then also operate on each field differently, depending on the datatype. If we want the flexibility for that, happy to implement. 

I also don't know how effective some of the other fields are, at least as far as it relates to subscription filtering. There may be other uses I'm not aware of though

Reply:
so attributes are preconfigured in like the AWS console?  Or they're determined at runtime? 

> So we would have to split on the string, then also operate on each field differently, depending on the datatype.

Yeah, I think that's fine?  It's maybe a big uglier in maxwell code but it appears cleaner to the end user.  Or if you've got a better idea I'm all ears, I just want to avoid having to add a new cmdline flag for each attribute type.

> I also don't know how effective some of the other fields are, at least as far as it relates to subscription filtering.

I think it's fine if we only support database/table/ts for now (those are the hyper obvious ones), as long as our interface leaves some expansion room down the line. 

> There may be other uses I'm not aware of though.

In my experience at *some* point if enough people adopt the code someone will want to be doing shit like extracting data out of a row and passing it as a message attribute.  but that's still hypothetical for now.

Reply:
Attributes configuration has been updated to use `sns_attrs` instead, and the respective documentation was updated as well.

It might be my minimal Java experience, but I was having a real hard time accessing the `RowMap` properties dynamically. I met somewhere in the middle, and allow only `database` and `table` to be added, but the producer can be updated to handle more values in `sns_attrs` as needed. 

It would support passing some of the `data` / `old_data` dynamically as attributes, but that would probably require some large scope changes to the `RowMap`. Thoughts?